CMakeLists.txt

```diff
@@ -736,6 +730,17 @@ ELSE()
   MESSAGE(STATUS " PARPACK library: Using internal Elmer library")
   MESSAGE(STATUS "------------------------------------------------")
 ENDIF()
+IF(WITH_Zoltan AND NOT USE_SYSTEM_ZOLTAN)
+  MESSAGE(STATUS " Zoltan: building bundled code")
+  MESSAGE(STATUS "------------------------------------------------")
+ELSEIF(WITH_Zoltan AND HAVE_ZOLTAN)
+  MESSAGE(STATUS " Zoltan library: " "${ZOLTAN_LIBRARY}")
+  MESSAGE(STATUS " Zoltan headers: " "${ZOLTAN_INCLUDE_DIR}")
+  MESSAGE(STATUS "------------------------------------------------")
+ELSEIF(WITH_Zoltan AND NOT ZOLTAN_FOUND)
+  MESSAGE(STATUS " Zoltan system library not found")
```

Comment:
Should this be a `MESSAGE(FATAL_ERROR ...)` instead of a `MESSAGE(STATUS ...)`.
If I understand correctly, we reach here if the user set to build with Zoltan (using the libraries installed on the build system) but it couldn't be found. That probably means that the user still needs to provide more configuration flags. Or something else is inconsistent...

Reply:
From the current code version, I got the impression that the compilation could (or should?) continue even if the library was not found; that's why I left the last condition and allowed it to continue.
It may be better to stop the configuration to alert the user of the error when locating the library and correct the problem. I will make sure to update the code accordingly.

Reply:
You are right.

IIUC, the current situation is a bit inconsistent. E.g., for `WITH_Trilinos=ON`, `WITH_MUMPS=ON` and some others, the configuration would fail if the corresponding packages wouldn't be found. On the other hand, for `WITH_Zoltan=ON`, `WITH_MMG=ON` ans some others, the configuration would succeed (with a status message hinting at the missing dependency) if those packages wouldn't be found.

But I agree with you that it is probably out of the scope for this PR to straighten these inconsistencies out.

Sorry for the noise.